chainer/variable.py

```diff
@@ -1792,18 +1808,19 @@ def initialize(self, shape):
 
         data = initializers.generate_array(
             self.initializer, shape, xp, device=device)
+        # Convert the array for iDeep.
+        # (generate_array produces numpy array even if device is Intel64)
+        # TODO(niboshi): This could be done in generate_array().
+        if isinstance(device, intel64.Intel64Device):
+            data = device.send_array(data)
```

Comment:
It converts `data` to `ideep.mdarray` before `self.array = data` because now `array()` rejects a numpy array if `device is Intel64Device`.